plugin-server/src/main/ingestion-queues/session-recording-v2/consumer.ts

```diff
@@ -277,6 +303,6 @@ export class SessionRecordingIngester {
 
         this.metrics.resetSessionsHandled()
 
-        return Promise.resolve()
+        await this.sessionBatchManager.flush()
```

Comment:
This reads like "when my partition is revoked I will flush" - I don't think this will work. You can't commit the offset if your assignment is revoked.

Reply:
You're right, I was trying to match the behaviour with the old ingester, but I forgot it was discarding the data for the revoked partitions – will fix that.

Reply:
Ok, so I read up on this and you can actually commit offsets when partitions are being revoked, according to the librdkafka docs:

https://docs.confluent.io/platform/current/clients/librdkafka/html/classRdKafka_1_1RebalanceCb.html

However, to keep things simpler for now, I'll change the behaviour to discard the data for the revoked partitions. We can revise the rebalancing algorithm later if we need to.